### PR TITLE
refactor: :building_construction: Restructure & rename packages

### LINKED ISSUE #1069: Restructure packages
```
 @digdir/design-system-react -> @digdir/designsystemet-react
 @digdir/design-system-tokens -> @digdir/designsystemet-theme
```

### PR DESCRIPTION
**Pull request by mimarz**
resolves #1069 

# Usage
New way of using packages will now be: 
```jsx
import '@digdir/designsystemet-theme'; // defaults to digdir theme
import '@digdir/designsystemet-css'; 
import { Button } from '@digdir/designsystemet-react';

<Button variant='secondary'>I am a button!</Button>;
```

# Summary of changes
- `@digdir/design-system-tokens` renamed to `@digdir/designsystemet-theme`
- `@digdir/design-system-react` now only contains legacy components.
  - Has `@digdir/designsystemet-react` as a dependecy and re-export all components to make transition away from legacy to new components easier. Will remove this when we reach v1.0.0.

- `@digdir/designsystemet-css` is now needed for correct styling on React components from `@digdir/designsystemet-react`

- `@digdir/designsystemet-react` now supports tree-shaking.

- `@digdir/designsystemet-react` extracts all css-modules to a single `react-components.css` which is exported as part of `@digdir/designsystemet-css`.
  - Tested using `Alert` for moving away from css-modules using only named classNames.

- Added new recommended extension [HTML-CSS-Class-Completion](https://github.com/zignd/HTML-CSS-Class-Completion) for intellisens on named classNames (such as `css/alert.css`)

# Test
```sh
npm i @digdir/designsystemet-react @digdir/designsystemet-theme @digdir/designsystemet-css @digdir/design-system-react@next
```

### CODE REVIEW

**Comment by mimarz** [comment]
@Barsnes Please have a look at the README's and if other stuff makes sense. We'll release a new prerelease tomorrow for testing downstream once we get #1641 and #1632 released.

File: README.md [hunk 1]
```diff
@@ -31,9 +31,17 @@ Our goal is to create consistent and user-friendly experiences in digital soluti
...
+[`@digdir/designsystemet-theme`](https://www.npmjs.com/package/@digdir/designsystemet-theme) - Themes for Designsystemet styling and components.
 
-[`@digdir/design-system-tokens`](https://www.npmjs.com/package/@digdir/design-system-tokens) – Tokens that contain UI data like colors and spacing for building user interfaces.
+[`@digdir/designsystemet-css`](https://www.npmjs.com/package/@digdir/designsystemet-css) - Styling for components
+
+[`@digdir/designsystemet-react`](https://www.npmjs.com/package/@digdir/designsystemet-react) - React implementation of Designsystemet components
+
+### Deprecated
+
+[`@digdir/design-system-react`](https://www.npmjs.com/package/@digdir/design-system-react) – Replaced by `@digdir/designsystemet-react`. Contains legacy components which are no longer maintained
```
Comment (Barsnes) [inline README.md:42]: that are no longer maintained

File: README.md [hunk 2]
```diff
@@ -43,17 +51,31 @@ Follow these steps to get started with the React components.
...
+```sh
+npm i typescript --save-dev
 ```
 
-### 2. Add the Inter font
+### 2. Font
+
+You are free to use any font-family with the components.
+
+The components are designed and developed using the [Inter font](https://github.com/rsms/inter) so variantions might occur if a different font is used.
```
Comment (Barsnes) [inline README.md:70]: variantions -> variations

File: README.md [hunk 3]
```diff
@@ -43,17 +51,31 @@ Follow these steps to get started with the React components.
...
+
+#### Add the Inter font (optional)
 
 Add the `<link>` tag to your application and set the `font-family` to `Inter`.
```
Comment (Barsnes) [inline README.md:74]: Add the `<link>` tag in `<head>`, and set the `font-family` to `Inter` in your global css file.

**Review by Barsnes — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (mimarz) [reply README.md:74]: Changed to 
> Add the `<link>` tag in `<head>`, and set `font-family` to `Inter` in your global css file.

File: README.md [hunk 4]
```diff
@@ -76,14 +98,15 @@ If you choose to install the font in a different way, remember to include the `4
 ### 3. Use a React component
 
 ```jsx
-import '@digdir/design-system-tokens/brand/digdir/tokens.css';
+import '@digdir/designsystemet-theme/brand/digdir/tokens.css';
+import '@digdir/designsystemet-css'; // Must be imported after @digdir/designsystemet-theme
```
[the remark this answers is not in the document]
Comment (mimarz) [reply README.md:102]: '@digdir/designsystemet-css' uses css variables declared in `@digdir/designsystemet-theme/brand/digdir/tokens.css` so yes?

File: packages/react-old/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+### 3. Use a React component
+
+```jsx
+import '@digdir/designsystemet-theme/brand/digdir/tokens.css';
+
+import { Button } from '@digdir/design-system-react';
+
+<Button variant='secondary'>I am a button!</Button>;
+```
```
[the remark this answers is not in the document]
Comment (mimarz) [reply packages/react-old/README.md:65]: This is still true. As mentioned in PR summary:

> - @digdir/design-systemet-react now only contains legacy components.
> 
>   - Has @digdir/designsystemet-react as a dependecy and re-export all components to make transition away from legacy to new components easier. Will remove this when we reach v1.0.0.

File: packages/react-old/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+If you use Typescript, make sure you have typescript >= 3.8 as a devDependency:
+
+```sh
+npm install typescript --save-dev
+```
+
+#### Legacy components
+
+If you are still using components prefix with `Legacy` you will also need to install the old tokens package.
```
[the remark this answers is not in the document]
Comment (mimarz) [reply packages/react-old/README.md:25]: Good suggestion

File: packages/react/README.md [hunk 7]
```diff
@@ -1,67 +1,12 @@
...
-React components from the DigDir Common Designsystem
+React implementation of the Designsystemet components
 
-## 🚀 Get started
+- Headless components.
+  - Uses `@digdir/designsystemet-theme` and `@digdir/designsystemet-css` for styling.
+- All components support `forwardRef`.
+- Most components extend and behave as native html-elements.
+- Most components support composition and `asChild` for overriding the underlying html-element.
+- React Server Components support.
```
[the remark this answers is not in the document]
Comment (mimarz) [reply packages/react/README.md:10]: Yeah, we can just update as we got along and get to test things properly :)

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Barsnes) [reply README.md]: You can declare a css variable after it is used, so the order of importing should not matter here

[the remark this answers is not in the document]
Comment (mimarz) [reply README.md]: works either way. Updated example

**Review by Barsnes — APPROVED** [review]
🤞🏻

**Pull request merged**